src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubEnvAction.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2019 Tim Jacomb
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.jenkinsci.plugins.github_branch_source;
+
+import hudson.model.InvisibleAction;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+/**
+ * Caches SCM values for a Run
+ */
+@Restricted(NoExternalUse.class)
+public class GitHubEnvAction extends InvisibleAction {
```

Comment:
Redundant class. Information is already retained the in the SCMRevision assoicated with the branch

Reply:
Yes but the javadoc for `EnvironmentContributor` says to use an `Action` to cache data that can be expensive to calculate, and it's expensive due to API calls required if you have to look it up